### DISCUSSION TITLE
Input for slug generation

### DISCUSSION THREAD

**Discussion by rlightner** [beholdr/maska#109] [Q&A]
Trying to figure out the mask to support a user typing in a mask.

Spaces will be converted to _ and all text will be lowercased

Any other character other than A-Z, a-z, 0-9, and _ will be ignored.

Thoughts? Thanks!

**Comment by beholdr** [comment]
You could use `preProcess` hook for this:

``` html
<script setup>
const options = {
  preProcess: (val) => val.toLowerCase().replace(' ', '_')
}
</script>

<template>
  <input v-maska:[options] data-maska="A" data-maska-tokens="A:[_0-9a-zA-Z]:multiple">
</template>
```
